### PR TITLE
show core team status in user profile

### LINKED ISSUE #11418: Core team status on user profiles
We plan to move archive our [core team Gitter chat room](https://gitter.im/freecodecamp/coreteam) and move our core team over to our new [contributor chat room](https://gitter.im/freecodecamp/contributors) - which is open to anyone who wants to contribute to our open source community through:
- open source codebase contributions
- YouTube videos
- Medium or Wiki articles
- Events through our Campsites
- curriculum development, and more

Instead of communicating core team status through presence in our core team room, core team members will instead have a special text on their Free Code Camp profile, sort of like this:

Core team membership will be on a year-by-year basis, so core team contributors who get busy and don't have time to contribute any more will automatically be recognized as past core team members, while ones who stay on will continue to accumulate annual core team distinctions.

Here is what we need to do:
- [ ] create an array attribute coreTeam on the User model, defaulting to []
- [ ] update the profile view to show the text "Core Team N" where N is the year for each year that a camper is on the core team. For example:

```
coreTeam: ["2015","2016"]
```

would show the text:
Core Team 2016
Core Team 2015

in reverse chronological order, below earned certificates.

Once this pull request has been accepted, I will manually go into the production database and set this status to true for core team members.

### PR DESCRIPTION
**Pull request by ivantedja**
#### Pre-Submission Checklist

- [x] Your pull request targets the `staging` branch of FreeCodeCamp.
- [x] Branch starts with either `fix/`, `feature/`, or `translate/` (e.g. `fix/signin-issue`)
- [x] You have only one commit (if not, [squash](http://forum.freecodecamp.com/t/how-to-squash-multiple-commits-into-one-with-git/13231) them into one commit).
- [x] All new and existing tests pass the command `npm run test-challenges`. Use `git commit --amend` to amend any fixes.
#### Type of Change

- [ ] Small bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Add new translation (feature adding new translations)
#### Checklist:

- [x] Tested changes locally.
- [x] Closes currently open issue (replace XXXX with an issue no): Closes #11418

### CODE REVIEW

**Comment by raisedadead** [comment]
Would this look cleaner:

image

than current:

image

**Comment by raisedadead** [comment]
It would be cleaner with badges. And a setting to manage the visibility of them as well.

**Comment by ivantedja** [comment]
That looks better, I'll update it. Badge looks more beautiful, I think it will be better if we have different kind of badge each year (more eye catching) but I'm not sure who will create the asset for that

**Comment by QuincyLarson** [comment]
@ivantedja rather than have a static badge image, perhaps you could one using built using CSS, so we could dynamically change the year?

File: common/models/user.json [hunk 1]
```diff
@@ -206,6 +206,10 @@
       "type": "string",
       "description": "An IETF language tag",
       "default": "en"
+    },
+    "coreTeam": {
```
Comment (BerkeleyTrue) [inline common/models/user.json]: Let's change the structure to be more extensible.

```json
  badges: {
    coreTeam: [
      "2016"
    ]
  } 
```

This way we can add more badges in the future

**Review by BerkeleyTrue — CHANGES_REQUESTED** [review]

File: common/models/user.json (hunk 1 above)
Comment (atjonathan) [inline common/models/user.json]: :+1:

**Comment by Mongues** [comment]
t

On Oct 29, 2016 2:13 AM, "Jonathan" <notifications@github.com> wrote:
>
> @atjonathan commented on this pull request.
>
> ________________________________
>
> In common/models/user.json:
>
> > @@ -206,6 +206,10 @@ "type": "string", "description": "An IETF language
tag", "default": "en" + }, + "coreTeam": {
>
> Hi👍
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
Am I on the new w/ John and Dave etc??

t

**Comment by raisedadead** [comment]
Hello!

Looks like you may have accidentally subscribed to all notifications for the [FreeCodeCamp](https://github.com/FreeCodeCamp/FreeCodeCamp) repository. To stop receiving notifications please visit the [subscriptions page](https://github.com/FreeCodeCamp/FreeCodeCamp/subscription) or click on the image below and select **"Not watching"**.

[image](https://github.com/FreeCodeCamp/FreeCodeCamp/subscription)


We are sorry for your inconvenience!

**Comment by Mongues** [comment]
Yeah but my team in here somewhere and by the way much respect as I've been
trying to learn from your thread following a long time please I have a few
quick questions please..

t

**Comment by ivantedja** [comment]
@QuincyLarson probably styling the shape can be done in separate task? It is possible to use CSS but my sense of art is not that great 😈 

@BerkeleyTrue I don't know why the travis is failed

**Comment by raisedadead** [comment]
@ivantedja build has been fixed, restarted it.

**Review by BerkeleyTrue — APPROVED** [review]

File: server/views/account/show.jade [hunk 2]
```diff
@@ -56,6 +56,10 @@ block content
                     if isBackEndCert
                         .button-spacer
                         a.btn.btn-primary.btn-block(href='/' + username + '/back-end-certification') View My Back End Development Certification
+    .row
+      .col-xs-12.text-center
+        if (badges.coreTeam && badges.coreTeam.length)
```
Comment (Bouncey) [inline server/views/account/show.jade]: Would this not be `badges.type.coreTeam`?

**Review by Bouncey — DISMISSED** [review]

File: server/views/account/show.jade (hunk 2 above)
Comment (ivantedja) [inline server/views/account/show.jade]: from my understanding, `type` is a keyword provided by `loopback` to specify the "schema" of the property

**Comment by raisedadead** [comment]
Tested.

#### :sparkles: :sparkles: **LGTM.** :sparkles: :sparkles: 

Thanks.

**Comment by BerkeleyTrue** [comment]
To be clear, `user.badges.coreTeam = ['2015', '2016']`. Is that correct?

**Comment by raisedadead** [comment]
Yup, in reverse order, though:

user.badges.coreTeam = ['2016', '2015']

**Comment by BerkeleyTrue** [comment]
Then this line `h4 Core Team #{badges.coreTeam.reverse().join(', ')}` will print in chronological order?

`Core Team 2015, 2016`

**Comment by raisedadead** [comment]
Yup.

**Comment by BerkeleyTrue** [comment]
Then why are we putting them in reverse order if we are just showing them in chronological?

**Comment by raisedadead** [comment]
Oh, sorry my bad. Yes you are correct, Quincy is supposed to enter the Years in the backend manually in the chronological order which you correctly said:

`user.badges.coreTeam = ['2015', '2016']`

and the output should now turn out like this:

image

**Comment by BerkeleyTrue** [comment]
Thanks @ivantedja!

**Comment by QuincyLarson** [comment]
@ivantedja great work!

**Pull request merged**